src/esp/bindings/SimBindings.cpp

```diff
@@ -138,6 +170,54 @@ void initSimConfigBindings(py::module& m) {
           R"(See See tutorials/async_rendering.py.)");
 }
 
+void initRenderInstanceHelperBindings(py::module& m) {
+  py::class_<RenderInstanceHelper, RenderInstanceHelper::ptr>(
+      m, "RenderInstanceHelper")
+      .def(py::init<sim::Simulator&, bool>(), py::arg("sim"),
+           py::arg("use_xyzw_orientations"), "todo")
+      .def("add_instance", &RenderInstanceHelper::AddInstance,
+           py::arg("asset_filepath"), py::arg("semantic_id"), "todo")
+      .def("clear_all_instances", &RenderInstanceHelper::ClearAllInstances,
+           "todo")
+      .def("get_num_instances", &RenderInstanceHelper::GetNumInstances, "todo")
+      .def(
+          "set_world_poses",
+          [](RenderInstanceHelper& self, py::array_t<float> positions,
+             py::array_t<float> orientations) {
+            // Get the number of instances from the helper
+            size_t num_instances = self.GetNumInstances();
+
+            // Validate the shape of positions
+            if (positions.ndim() != 2 || positions.shape(0) != num_instances ||
+                positions.shape(1) != 3) {
+              throw std::runtime_error(
+                  "positions must be a 2D array of shape (num_instances, 3).");
+            }
+
+            // Validate the shape of orientations
+            if (orientations.ndim() != 2 ||
+                orientations.shape(0) != num_instances ||
+                orientations.shape(1) != 4) {
+              throw std::runtime_error(
+                  "orientations must be a 2D array of shape (num_instances, "
+                  "4).");
+            }
+
+            if (!is_contiguous(positions) || !is_contiguous(orientations)) {
+              throw std::runtime_error(
```

Comment:
Is this error passed to Python?

Reply:
yes, std::runtime_error becomes Python RuntimeError exception